CMakeLists.txt

```diff
@@ -329,7 +329,10 @@ if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
     if(MSVC)
         set(EXE_CFLAGS "${EXE_CFLAGS} /w")
     else()
-        set(EXE_CFLAGS "${EXE_CFLAGS} -Werror -Wall -Werror=implicit-fallthrough")
+        set(EXE_CFLAGS "${EXE_CFLAGS} -Werror -Wall")
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.0)
```

Comment:
I think this check needs to be more specifically targeted towards GCC. Clang has fallthrough stuff in much earlier versions.